server/src/main/java/org/opensearch/index/mapper/DocumentMapper.java

```
@@ -328,9 +331,39 @@ public void validate(IndexSettings settings, boolean checkLimits) {
                 );
             }
         }
+
+        // Indexing Sort with Nested Fields is only supported on & after Version 3.2.0
         if (settings.getIndexSortConfig().hasIndexSort() && hasNestedObjects()) {
-            throw new IllegalArgumentException("cannot have nested fields when index sort is activated");
+            if (settings.getIndexVersionCreated().before(Version.V_3_2_0)) {
+                throw new IllegalArgumentException("cannot have nested fields when index sort is activated");
+            }
+
+            /*
+             * Index sorting works for regular fields across documents that may contain nested objects,
+             * but sorting on fields inside nested objects is not supported. This validation checks
+             * the index sort configuration and throws an exception if any sort field is inside
+             * a nested object.
+             */
+            List<String> sortFields = settings.getValue(IndexSortConfig.INDEX_SORT_FIELD_SETTING);
+            for (String sortField : sortFields) {
+                Mapper mapper = this.fieldMappers.getMapper(sortField);
+                if (mapper != null && mapper.name().contains(".")) {
+                    String parentPath = mapper.name().substring(0, mapper.name().lastIndexOf('.'));
+                    ObjectMapper nestedParent = objectMappers().get(parentPath);
+                    if (nestedParent != null && nestedParent.nested().isNested()) {
```

Should we throw any exception if parent field is not found or its taken care of in the flow ?

---

that validation is getting handled by Lucene itself.

---

That test will not be required here right, as those validation are applied while parsing the mapping, in the InternalEngineTest we are already adding parsed documents to the Lucene index?